File: hello.rb
#!/usr/bin/env ruby
require 'fileutils'
version='%%VERSION%%-%%RELEASE%%'
def imcdate
    t = Time.now
    return t.strftime("%Y.%m.%d-%H.%M.%S") 
end

if ARGV[0] =~ /-v/i
    puts 'hello.py ' + version.to_s
    exit
end

myLogDir = ENV['HOME'] + '/log/hello.rb/' + imcdate

FileUtils.mkdir_p(myLogDir)
File.open(myLogDir+'/hello.log', 'w') do |f|
   f.puts imcdate + ' : INFO : ' + 'Hello World!'
end 
puts 'Hello World!'

File: hello.py
#!/usr/bin/env python
import sys
import datetime
import os
import re

version = '%%VERSION%%-%%RELEASE%%'

    
def imcdate():
    now = datetime.datetime.now()
    return now.strftime("%Y.%m.%d-%H.%M.%S")

def main():
    if len(sys.argv) != 1:
        if re.search(r'-v', sys.argv[1], re.I):
            print 'hello.py ' + str(version)
            return 0

    myHome = os.environ.get('HOME')
    myLogDir = myHome + '/log/hello.py/' + imcdate()
    if not os.path.exists(myLogDir):
        os.makedirs(myLogDir)

    LOG = open(myLogDir + '/hello.log', 'w')

    LOG.write(imcdate() + ' : INFO : ' + "Hello World!\n")
    print 'Hello World!'
    return 0

if __name__ == '__main__':
    sys.exit(main())
